docs/pages/workflow/prebuild.mdx

```diff
@@ -24,7 +24,16 @@ We highly recommend using prebuild for the reasons listed in the [pitch](#pitch)
 
 ### Usage with EAS Build
 
-You can configure EAS Build to run `npx expo prebuild` before building by [using the managed preset](/build-reference/ios-builds/).
+If your project does not contain **ios** and **android** folders, then EAS Build will run Prebuild to generate your native code just before compilation. This is the default for any project started from `npx create-expo-app`.
+
+If you have a project that contains **android** and **ios** folders, then EAS Build will not run Prebuild by default in order to avoid overwriting any changes you've made to your native code.
+
+If you sometimes run `npx expo prebuild`, `npx expo run:android`, or `npx expo run:ios` locally to troubleshoot your native code but still want EAS Build to continue running Prebuild to generate fresh native projects during build, then you should add the **android** and **ios** folders to your **.gitignore** or **.easignore** files:
```

Comment:
```suggestion
If you troubleshoot your app by [compiling it locally](/guides/local-app-development/#local-app-compilation) (running `npx expo prebuild`, or `npx expo run:android` or `npx expo run:ios`), you can still use Prebuild with EAS Build to generate fresh native directories during the build process. In this scenario, add the **android** and **ios** directories to **.gitignore** or **.easignore** files:
```

- I think its important to add the link to the local app compile section as now we have that as to what it means to compile it locally
- I've also rephrased it a bit as to emphasize the subject here (as to which use case people might want to compile their app locally when using EAS Build)